test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
> I think we need to implement an approach when Container is cleared or re-created before each individual test.

I think you are right. But the question shall it be a public `clear` method or shall we dynamically reset all properties this way: `(Container as any)["instances"] = (Container as any)["paramHandlers"] = ... = [];`